File: 剑指offer/10.1.斐波那契数列.py
class Solution:
    def fib(self, n: int) -> int:
        if n == 0:
            return 0

        if n == 1:
            return 1

        ans = 0
        pre0 = 0
        pre1 = 1

        for _ in range(2, n + 1):
            ans = pre0 + pre1
            pre0 = pre1
            pre1 = ans

        return ans % 1000000007

File: 剑指offer/10.1.斐波那契数列.cpp
class Solution {
public:
    int fib(int n)
    {
        if (n <= 1) {
            return n;
        }

        int ans = 0;
        int pre1 = 0;
        int pre2 = 1;

        for (int i = 2; i <= n; i++) {
            ans = (pre1 + pre2) % 1000000007;

            pre1 = pre2;
            pre2 = ans;
        }

        return ans % 1000000007;
    }
};